src/controllers/expenses.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+const { expensesId } = require('../helpers/getId');
+
+const getAll = (expenses) => {
+  return (req, res) => {
+    const { userId, categories, from, to } = req.query;
+
+    let filteredExpenses = expenses;
+
+    if (!isNaN(userId)) {
+      filteredExpenses = filteredExpenses.filter(
+        (expense) => expense.userId === +userId
+      );
+    }
+
+    if (categories) {
+      const arrayCategory = Array.isArray(categories)
+        ? categories.forEach((category) => category.toLowerCase())
+        : [categories.toLowerCase()];
+
+      filteredExpenses = filteredExpenses.filter((expense) =>
+        arrayCategory.includes(expense.category.toLowerCase())
+      );
+    }
+
+    if (from) {
+      filteredExpenses = filteredExpenses.filter(
+        ({ spentAt }) => (new Date(spentAt)) > (new Date(from))
+      );
+    }
+
+    if (to) {
+      filteredExpenses = filteredExpenses.filter(
+        ({ spentAt }) => (new Date(spentAt)) < (new Date(to))
+      );
+    }
+
+    res.send(filteredExpenses);
+  };
+};
+
+const add = (expenses, users) => {
+  return (req, res) => {
+    const { userId, spentAt, title, amount, category, note } = req.body;
+
+    if (!userId || !spentAt || !title || !amount || !category) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const foundedUser = users.find((user) => user.id === +userId);
+
+    if (!foundedUser) {
+      res.sendStatus(400);
+
+      return;
+    }
+
+    const expense = {
+      id: expensesId.getId(),
+      ...req.body,
```

Comment:
Remove 'note' property from object since it already exists in 'req.body' which is spread